rules/jsx-import.js

```diff
@@ -0,0 +1,126 @@
+	React: ['react'],
+	preact: ['preact']
+};
+
+const pragmaAPIs = {
+	React: ['cloneElement']
+};
+
+const getSuggestedPackages = modules => arrayToSentence(modules.map(x => `\`${x}\``), {lastSeparator: ' or '});
+
+const hasPackage = (node, packages) => node.init ?
+	packages.some(x => x === astUtils.getRequireSource(node.init)) :
+	packages.some(x => x === node.parent.source.value);
+
+const isImportDeclaration = node => node.init ?
+	astUtils.isStaticRequire(node.init) :
+	node.parent.type === 'ImportDeclaration';
+
+const getVariable = (name, variables) => variables.find(x => x.name === name);
+
+// https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/util/variable.js#L27-L52
+const getVariablesInScope = context => {
+	let scope = context.getScope();
+	let {variables} = scope;
+
+	while (scope.type !== 'global') {
+		scope = scope.upper;
+		variables = variables.concat(scope.variables);
+	}
+
+	if (scope.childScopes.length > 0) {
+		variables = scope.childScopes[0].variables.concat(variables);
+
+		if (scope.childScopes[0].childScopes.length > 0) {
+			variables = scope.childScopes[0].childScopes[0].variables.concat(variables);
+		}
+	}
+
+	return variables.reverse();
+};
+
+const create = context => {
+	let isJsx = false;
+	let isModuleAPIsCalled = false;
+	let pragma;
+
+	return {
+		Program: () => {
+			for (const key of Object.keys(pragmas)) {
+				pragma = pragma || getVariable(key, getVariablesInScope(context));
+			}
+		},
+		CallExpression(node) {
+			const {callee} = node;
+			if (
+				pragma &&
+				callee.type === 'MemberExpression' &&
+				callee.object.name === pragma.name &&
+				pragmaAPIs[pragma.name].includes(callee.property.name)
+			) {
+				context.markVariableAsUsed(pragma.name);
+				isModuleAPIsCalled = true;
+			}
+		},
+		JSXOpeningElement: node => {
+			isJsx = true;
+
+			if (pragma) {
+				context.markVariableAsUsed(pragma.name);
+			} else {
+				context.report({
+					node,
+					message: 'A valid pragma must be in scope when using JSX'
+				});
+			}
+		},
+		'Program:exit': () => {
+			if (pragma) {
+				const {node} = pragma.defs[0];
+				const isDeclaration = isImportDeclaration(node);
+				const packages = pragmas[pragma.name];
+				const hasPkg = hasPackage(node, packages);
+
+				if (!isModuleAPIsCalled && !isJsx) {
+					context.report({
+						node,
+						message: `\`${pragma.name}\` shouldn't be imported when not using JSX`
+					});
+				}
+
+				if (isJsx && (!isDeclaration || !hasPkg)) {
+					context.report({
+						node,
+						message: `\`${pragma.name}\` should be imported from ${getSuggestedPackages(packages)}`,
+						fix: fixer => {
+							if (isDeclaration && packages.length === 1) {
+								const val = node.init ? node.init.arguments[0] : node.parent.source;
+								return fixer.replaceText(val, `'${packages[0]}'`);
+							}
+						}
+					});
+				}
+
+				isJsx = false;
+				isModuleAPIsCalled = false;
+			}
+		}
+	};
+};
+
+module.exports = {
+	create,
+	meta: {
```

Comment:
You need to mark the type of the rule: https://github.com/sindresorhus/eslint-plugin-unicorn/commit/610a9014ab6a7076eaebf1990a143f15668ccba2